libsql/internal/http/shared/statement.go

```diff
@@ -32,13 +32,16 @@ func ParseStatement(sql string) ([]string, []ParamsInfo, error) {
 }
 
 func ParseStatementAndArgs(sql string, args []driver.NamedValue) ([]string, []Params, error) {
+	stmts, _ := sqliteparserutils.SplitStatement(sql)
+
+	if len(args) == 0 {
+		return stmts, nil, nil
```

Comment:
Can we return array for params here too: `return stmts, make([]Params, len(stmts)), nil`?
It's a bit of overhead but it should be pretty negligible, no? But the benefit is that API will be "uniform" and there will be less special cases in the dependent code...

Reply:
For 1M statements this array will occupy at most 32MB. But from what I saw locally - dump process consumes GBs of RAM to handle 1M statements or so...

So, I agree that this is not great, but I doubt if this is actually that critical...
I would rather keep API uniform but also work on truly streaming approach (but this will require more work...)

Reply:
Streaming is my next step. Just fixing stuff on the way.

Reply:
I agree to some extend but this is not only about memory. Without special casing the arguments handling code will execute on those empty arrays and it does some logic including conditions. I don't want to execute this for 1M statements when I don't have to.